File: leetcode/541.reverse-string-ii/go/reversestr_test.go
package reversestr

import "testing"

type testcase struct {
	s    string
	k    int
	want string
}

func TestReverseStr(t *testing.T) {
	tcs := []testcase{
		{s: "abcd", k: 2, want: "bacd"},
		{s: "abcdefg", k: 2, want: "bacdfeg"},
	}

	for _, tc := range tcs {
		got := reverseStr(tc.s, tc.k)
		if got != tc.want {
			t.Fatalf("reversestr, want: %s, got: %s", tc.want, got)
		}
	}
}

File: leetcode/541.reverse-string-ii/ts/reverseStr.test.ts
import { test, expect } from "vitest"
import reverseStr from "./reversestr.js"

interface testcase {
    s: string
    k: number
    want: string
}

test("reversestr", () => {
    const tcs: testcase[] = [
        { s: "abcd", k: 2, want: "bacd" },
        { s: "abcdefg", k: 2, want: "bacdfeg" },
    ]

    tcs.forEach(tc => expect(reverseStr(tc.s, tc.k) === tc.want))
})
